docs/05-actor-model/03-actor-in-blokchain.md

```diff
@@ -0,0 +1,440 @@
+a look at it.
+
+There are three types on it, let's go through them one by one.
+The first one is an `InstantiateMsg`. This is the one, which is sent
+on contract instantiation. It typically contains some data which
+are needed to properly initialize it. In most cases, it is just a
+simple structure.
+
+Then there are two enums: `ExecuteMsg`, and `QueryMsg`. They are
+enums because every single variant of them represents a different
+message which can be sent. For example the `ExecuteMsg::UpdateAdmin`
+corresponds to the `update_admin` message we were sending previously.
+
+Note, that all the messages are attributed with
+`#[derive(Serialize, Deserialize)]`, and
+`#[serde(rename_all="snake_case")]`. Those attributes come from
+the [serde](https://serde.rs/) create, and they help us with
+deserialization of them (and serialization in case of sending
+them to other contracts). The second one is not required,
+but it allows us to keep a camel-case style in our Rust code,
+and yet still have JSONs encoded with a snake-case style more
+typical to this format.
+
+I encourage you to take a closer look at the `serde` documentation,
+like everything it is there, can be used with the messages.
+
+One important thing to notice - empty variants of those enums,
+tends to use the empty brackets, like `Admin {}` instead of
+more Rusty `Admin`. It is on purpose, to make JSONs cleaner,
+and it is related to how `serde` serializes enum.
+
+Also worth noting is that those message types are not set in stone,
+they can be anything. This is just a convention, but sometimes
+you would see things like `ExecuteCw4Msg`, or similar. Just keep
+in mind, to keep your message name obvious in terms of their
+purpose - sticking to `ExecuteMsg`/`QueryMsg` is generally a good
+idea.
+
+## Entry points {#entry-points}
+
+So now, when we have our contract message, we need a way to handle
+them. They are send to our contract via entry points. There are
+three entry points in the `cw4-group` contract:
+
+```rust
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    // ...
+}
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    // ..
+}
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    // ..
+}
+```
+
+Those functions are called by the CosmWasm virtual machine when
+a message is to be handled by contract. You can think about them
+as the `main` function of normal programs, except they have a signature
+that better describes the blockchain itself.
+
+What is very important, is that the names of those entry points (similarly to
+the `main` function) are fixed - it is relevant, so the virtual machine knows
```

Comment:
```suggestion
the `main` function) are static - it is relevant, so the virtual machine knows
```

Reply:
I don't want to use a `static` world in particular here. Because of C++ influence, `static` have a very particular meaning for functions in software engineering, and it may be misunderstood.